demo/admin/src/products/ProductForm.tsx

```diff
@@ -71,17 +84,32 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
         id ? { variables: { id } } : { skip: true },
     );
 
-    const initialValues: Partial<FormValues> = data?.product
-        ? {
-              ...filterByFragment<GQLProductFormManualFragment>(productFormFragment, data.product),
-              image: rootBlocks.image.input2State(data.product.image),
-          }
-        : {
-              image: rootBlocks.image.defaultValues(),
-              inStock: false,
-              additionalTypes: [],
-              tags: [],
-          };
+    const initialValues: Partial<FormValues> = React.useMemo<Partial<FormValues>>(() => {
```

Comment:
because I need to access fetched data after calling filter() multiple times and I remembered the same situation somewhere else where somebody recommended wrapping in useMemo and preparing data in the callback function. :-) 

ah, thats even current best-practice provided by admin-form-generator. 👍